feat(KL-125): get all products

ohhamma opened the pull request:
## 📌 연관된 이슈
- **[KL-125/전체 상품 목록 조회](https://ohhamma.atlassian.net/browse/KL-125)**

## 📝 작업 내용
- 모든 상품 목록 조회 API 구현
- 페이지네이션 적용
- 불필요한 Entity 키워드 제거

## 🌳 작업 브랜치명
- `KL-125/전체-상품-목록-조회`

## 📸 스크린샷 (선택)

## 💬 리뷰 요구사항 (선택)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
  - Introduced paginated retrieval for product listings.
  - Simplified response handling for product creation, updates, and deletions.

- **Bug Fixes**
  - Renamed methods for consistency in the product retrieval process.

- **Tests**
  - Expanded test coverage for product retrieval and pagination scenarios.
  - Updated test method names to align with recent changes in the product service.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ohhamma committed:
KL-125/style: rename method

- id로 상품을 가져오는 메소드 이름 getProductById로 변경

ohhamma committed:
KL-125/feat: get all products service

- 모든 상품 목록 조회하는 비즈니스 로직 구현

ohhamma committed:
KL-125/test: get all products service

- 모든 상품 목록 조회 비즈니스 로직 테스트
- 테스트 상품 상수 하나 추가

ohhamma committed:
KL-125/feat: get all products api

- 모든 상품 목록 조회 API 구현

ohhamma committed:
KL-125/test: get all products api

- 모든 상품 목록 조회 API 테스트
- 클래스 변수명 변경

ohhamma committed:
KL-125/feat: add pagination

- 페이지네이션 추가
  - 기본 페이지 사이즈 값, 페이지 넘버 설정
- 페이지 관련 기본값들 상수로 추가
- 관련 테스트코드 작성

idealflower-k commented on src/main/java/taco/klkl/domain/product/controller/ProductController.java:
```diff
@@ -28,40 +35,42 @@ public class ProductController {
-		@PathVariable Long id
-	) {
+	@ResponseStatus(HttpStatus.NO_CONTENT)
+	public void deleteProduct(@PathVariable Long id) {
```
delete시 NoContent응답으로 반환시 GlobalResponse내용도 없이 나가는게 맞나요?

min3m replied:
이 부분 GlobalResponse로 wrapping이 필요할거 같습니다!

ohhamma replied:
이 부분을 어떻게 할 지 고민이 많았습니다..
이전에 `EntityResponse`를 사용해서 `noContent()`로 `GlobalResponse`로 감싸서 나가되, `data` 부분이 비워져있는 채로 보냈었는데, `ResponseStatus`를 아에 `204`로 설정해버리니 값이 안나가더라고요.

우선은 일관성 있게 `GlobalResponse`로 감싸도록 다시 수정해놓겠습니다!

idealflower-k commented on src/main/java/taco/klkl/domain/product/service/ProductService.java:
```diff
@@ -21,7 +25,13 @@ public class ProductService {
 	private final UserUtil userUtil;
 
-	public ProductDetailResponseDto getProductInfoById(final Long id) {
+	public List<ProductSimpleResponseDto> getAllProducts(Pageable pageable) {
```
`Pageable`은 어떤 객체인가요?

ohhamma replied:
페이징 정보를 포함한 객체입니다! 페이지 번호와 페이지 크기를 담고 있는 객체라고 생각하시면 됩니다.

min3m commented on src/test/java/taco/klkl/domain/product/service/ProductServiceTest.java:
```diff
@@ -41,6 +44,72 @@ void setUp() {
+		// Then
+		assertThat(result).hasSize(2);
+
+		assertThat(result.get(0))
```
extracting으로 값을 추출해서 containsExactly로 비교하는 과정인 건가요?? assertThat 내에서 하나하나 비교하는 것보다 가독성이 훨씬 좋은것 같습니다!

ohhamma replied:
맞습니다! 저도 이 방법은 처음 사용해보는건데, 가독성 측면에서 정말 좋더라고요.

min3m reviewed (approved):
@RequestParam 에서 Default 값 설정하여 이용하는 부분 덕분에 잘 봤습니다! 
PR 공유해주셔서 정말 감사합니다~.
그 외에 궁금한 점이 있는 부분은 코멘트 달아놨습니다! 한번 확인 부탁드립니다.

ohhamma committed:
KL-125/fix: delete product api

- 상품 삭제 API 응답도 GlobalResponse로 감싸도록 수정
- 이에 따라 테스트코드 수정
- 가독성을 위해 컨트롤러 파라미터에 개행 추가

ohhamma commented:
@coderabbitai review
